app/controllers/katello/api/v2/activation_keys_controller.rb

```diff
@@ -197,13 +197,28 @@ def remove_subscriptions
 
     api :PUT, "/activation_keys/:id/content_override", N_("Override content for activation_key")
     param :id, :identifier, :desc => N_("ID of the activation key"), :required => true
-    param :content_override, Hash, :desc => N_("Content override parameters") do
+    param :content_override, Hash, :desc => N_("Content override parameters"), :deprecated => true do
+      param :content_label, String, :desc => N_("Label of the content"), :required => true
+      param :value, [0, 1, "default"], :desc => N_("Override to 0/1, or 'default'"), :required => true
+    end
+    param :content_overrides, Array, :desc => N_("Array of Content override parameters") do
       param :content_label, String, :desc => N_("Label of the content"), :required => true
       param :value, [0, 1, "default"], :desc => N_("Override to 0/1, or 'default'"), :required => true
     end
     def content_override
-      content_override = validate_content_overrides(params[:content_override])
-      @activation_key.set_content_override(content_override[:content_label], 'enabled', content_override[:value])
+      if params[:content_override]
+        ::Foreman::Deprecation.api_deprecation_warning("The parameter content_override will be removed in Katello 3.4. Please update to use the uploads parameter.")
+      end
+      content_overrides = params[:content_overrides] ? params[:content_overrides] : [params[:content_override]]
+      content_overrides = [] unless content_overrides
+      content_override_values = content_overrides.map do |content_override_params|
+        validate_content_overrides(content_override_params)
+      end
+
+      content_override_values.each do |content_override|
+        @activation_key.set_content_override(content_override[:content_label], 'enabled', content_override[:value])
```

Comment:
Do we have to make a call to Candlepin for each provided override or can they be sent in bulk?